text/0000-network-transform-buffered-interpolation.md

```diff
@@ -0,0 +1,408 @@
+```C#
+    public class MyOwnBufferedInterpolator<T> : IInterpolator<T> where T : struct
+    {
+        // ...
+        protected override double ServerTickBeingHandledForBuffering => ServerTick - OtherBuffering;
+```
+Or they could use `NoInterpolator` which just returns your latest position, without doing anything to it.
+
+## Other ways to reduce jitter
+
+Reducing the number of packets sent is a good way to prevent jitter, however the goal of this RFC is really to mitigate the jitter that does happen, even after mitigation.
+
+Increasing send rate will help reducing the buffer size needed for clean interpolation. With shorter ticks, you need to wait a shorter time before saying you can process a server position. This could be done through education when talking about these issues.
+
+## Other design
+The interpolator could have been hard coded in NetworkTransform without actually being a separate class. This could have made NetworkTransform more self contained, but would also have been more cumbersome to extend interpolation. Users could for example want to prevent interpolated transforms from going through walls instead of interpolating linearly. They could want to interpolate using their navmesh. Any sort of custom interpolation becomes easy with the current design and would hook directly in NetworkTransform, without being too hard to maintain on our side. 
+With .meta file default values, you can set a ScriptableObject as a default for your class, no extra user action should be needed to use NetworkTransform with default scriptable object values.
+
+## Impact of not doing this
+Buffered interpolation is simple on paper, but can be frustrating to implement, even for netcode experts. It's a recurring technique to allow resisting to bad network conditions like you can find on mobile platforms.
+Having this set as a default will allow giving users a NetworkTransform that is smooth by default, on a wide range of network conditions.
+
+## Why in NetworkTransform and not NetworkVariables
+
+Doing it directly in NetworkVariable right now isn't useful. Most of our use cases for interpolation are transform related. 
+
+Going the NetworkVariable route would make us create APIs we'd need to maintain for a few years, without having concrete use cases.
+
+## Prediction and interpolator.
+Current interpolator only affects ghosts and doesn't affect locally owned objects. An eventual predicted player wouldn't be affected by this interpolation.
+Future prediction might want to interpolate corrected values instead of teleporting players. This should be handled by prediction code.
+Since interpolator is a self contained class and isn't a MonoBehaviour, interpolators could be reused for that correction interpolation.
+
+## Other APIs for interpolator
+The Interpolator's `Update()` method could have been injected with the last 3 positions. 
+```C#
+Vector3 Update(Vector3 pos1, Vector3 pos2, Vector3 pos3)
```

Comment:
I prefer the current API, It allows to implement an interpolator which uses any number of data points. You could do 3 positions with the current API.